src/Main.java

```diff
@@ -0,0 +1,87 @@
+import java.util.ArrayList;
+import java.util.Random;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Pikachu pikachu =new Pikachu("Ash");
+        Charmander charmander =new Charmander("Ash");
+        Squirtle squirtle =new Squirtle("Ash");
+        Bulbasaur bulbasaur =new Bulbasaur("Ash");
+
+        Pokemon[] pokemons = {pikachu,charmander,squirtle,bulbasaur};
+
+        // pet them all
+        for( int i=0; i<pokemons.length; i++){
+            pokemons[i].pet();
+        }
+
+
+        // do the base attacks
+        System.out.println("\n\nWarmup time!");
+        pikachu.baseAttack(squirtle);
+        charmander.baseAttack(bulbasaur);
+        squirtle.baseAttack(charmander);
+        bulbasaur.baseAttack(pikachu);
+
+
+        for( int i=0;i<10; i++) {
+            healAll(pokemons);
+            battleRoyal(pokemons, pikachu, charmander, squirtle, bulbasaur);
+        }
+
+        //dump status of pokemons
+        for( int i=0; i<pokemons.length; i++){
+            System.out.println("\n"+pokemons[i]);
+        }
+    }
+
+
+    public static void healAll(Pokemon[] pokemons){
+        // heal them
+        System.out.println("\n\nHeal the pokemon!");
+        for( int i=0; i<pokemons.length; i++){
+            pokemons[i].heal();
+        }
+    }
+
+    public static void battleRoyal(Pokemon[] pokemons, ElectricPokemon pikachu, FirePokemon charmander, WaterPokemon squirtle, GrasPokemon bulbasaur){
```

Comment:
Leuke toepassing. Ik denk dat er nog wel het een en ander aan te verbeteren valt, maar het is goed om te zien dat je aan het experimenteren bent geslagen om iets moiis in elkaar te knutselen. Dit is, naar mijn mening, de beste manier om te leren programmeren